src/lib/network/network_autoyast.rb

```diff
@@ -210,32 +210,13 @@ def merge_routing(instsys_routing, ay_routing)
       instsys_routing.merge(ay_routing)
     end
 
-    # Returns current AY profile in the internal representation
-    #
-    # @return [Hash] hash representing current profile or empty hash
-    def ay_current_profile
-      Yast.import "Profile"
-
-      ay_profile = Profile.current
-
-      return {} if ay_profile.nil? || ay_profile.empty?
-
-      ay_profile
-    end
-
     # Returns networking section of current AY profile
     def ay_networking_section
-      return {} if ay_current_profile["networking"].nil?
-
-      ay_current_profile["networking"]
+      @ay_networking_section || {}
     end
 
-    # Returns global section of current AY profile
-    def ay_general_section
-      return {} if ay_current_profile["general"].nil?
+    attr_writer :ay_networking_section
```

Comment:
I guess we plan to use the Y2Network::AutoinstProfile::NetworkingSection object:

https://github.com/yast/yast-network/blob/master/src/lib/y2network/autoinst_profile/networking_section.rb#L37

Reply:
To be honest I am not sure and I do not want to break it more then it is already.

Reply:
Ok, we could do a intermediate step before adapting it completely.